### PR TITLE
Added implementations in MariaDB

### PR DESCRIPTION
**Pull request by fvbakel**
## Description

Added 3 implementations in MariaDB. See README.md for more details.

## Contributing requirements

* [X] I read the contribution guidelines in CONTRIBUTING.md.
* [X] I placed my solution in the correct solution folder.
* [X] I added a README.md with the right badge(s).
* [X] I added a Dockerfile that builds and runs my solution.
* [X] I selected `drag-race` as the target branch.
* [X] All code herein is licensed compatible with BSD-3.

### CODE REVIEW

**Comment by rbergen** [comment]
I think we will have to rename PrimeSQLite/solution_1 to PrimeSQL/solution_1, and make this PrimeSQL/solution_2. The language is (a) SQL (dialect) in both cases, it's the execution environment that differs.

**Comment by fvbakel** [comment]
@rbergen I agree. I refactored the SQL solutions to one directory now.

File: PrimeSQL/solution_2/README.md [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+![Parallelism](https://img.shields.io/badge/Parallel-no-green)
+![Bit count](https://img.shields.io/badge/Bits-32-yellowgreen)
+
+This solution consists of 3 implementations of the prime calculation in SQL using [MariaDB](https://mariadb.org). The characteristics of each implementation are described below.
+
+## Implementation details
+
+### Overview
+
+These implementations are based on the [SQLite implementation](../../PrimeSQL/solution_1/). The docker image that is used as a basis uses `docker-entrypoint.sh` as an entry point. This script has been modified to just run the sql scripts in this solution and then exit. In al cases memory stored tables without transaction support are used. To enable this and to avoid the table to be written to disk the setting `max_heap_table_size` was set to 1Gb.
```
Comment (rbergen) [inline PrimeSQL/solution_2/README.md]: Due to the rename of PrimeSQLite to PrimeSQL, the relative URL can be changed to `../solution_1/`.

File: PrimeSQL/solution_2/README.md [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+### MariaDB1 implementation
+
+This implementation uses the base algorithm. It makes use of a template table `primes_table_template`. This table has two columns. The first column (`n`) contains the natural numbers that are considered and the second column (`isPrime`) is always initialized to the value 1. In the beginning of the calculation this table is copied to a new table `primes_table`. The non prime numbers are eliminated on this table by setting the `isPrime` value to 0.
+
+For the `isPrime` column the datatype `INT` is used. This datatype uses 32 bits for storage. Various experiments revealed that different types like `TINYINT` and `BIT(1)` result in a worse performance.
+
+### MariaDB2 implementation
+
+This implementation does not use the base algorithm. It is as close as possible to the [SQLite solution 1](../../PrimeSQL/solution_1/) implementation and algorithm.
```
Comment (rbergen) [inline PrimeSQL/solution_2/README.md]: See earlier note on relative URL.

File: PrimeSQL/solution_2/go.sh [hunk 3]
```diff
@@ -0,0 +1,3 @@
+docker build --pull --rm -f Dockerfile -t mariadb:latest .
+#docker run --rm -it -v $PWD/tmp:/opt/app/ mariadb:latest
```
Comment (rbergen) [inline PrimeSQL/solution_2/go.sh]: I think it would be cleaner to remove commented-out debug/template/original code.

**Review by rbergen — COMMENTED** [review]
The Docker image output starts with quite a lot of lines that are related to the startup of the DBMS. Would it be possible to suppress those?

File: PrimeSQL/solution_2/README.md (hunk 1 above)
Comment (fvbakel) [reply PrimeSQL/solution_2/README.md]: Agree.

File: PrimeSQL/solution_2/README.md (hunk 2 above)
Comment (fvbakel) [reply PrimeSQL/solution_2/README.md]: Ok

File: PrimeSQL/solution_2/go.sh (hunk 3 above)
Comment (fvbakel) [reply PrimeSQL/solution_2/go.sh]: Agree, well spotted :-)

**Comment by fvbakel** [comment]
> The Docker image output starts with quite a lot of lines that are related to the startup of the DBMS. Would it be possible to suppress those?

Yes, that is possible. I refactored the running mechanism and reduced it to the bare minimum required to run the solutions.

**Comment by fvbakel** [comment]
Thanks for the review and suggestions. I think I have now implemented fixes for all comments

**Review by rbergen — APPROVED** [review]
LGTM

**Review by marghidanu — APPROVED** [review]

**Pull request merged**